pyproject.toml

```diff
@@ -64,6 +64,12 @@ docs = [
 Source = "https://github.com/matrix-nio/matrix-nio"
 Documentation = "https://matrix-nio.readthedocs.io/en/latest/"
 
+[tool.setuptools.packages.find]
+where = ["src"]
+
+[tool.setuptools.package-data]
+"nio" = ["py.typed"]
```

Comment:
I'm not 100% sure it's fully typed in a way that we wouldn't make downstream users sad... I'd love to be wrong

Reply:
for what i use in cyberbot, it works well :) and maybe encourages users to fix mistakes in nio when they see typing problems, so i'd suggest turning it on even if not perfect yet. 😎